SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -364,29 +368,41 @@ - (void)didEnterStateRegistered {
         [self sdl_transitionToState:SDLLifecycleStateUnregistering];
         return;
     }
-
     NSArray<SDLLanguage> *supportedLanguages = self.configuration.lifecycleConfig.languagesSupported;
-    SDLLanguage desiredLanguage = self.configuration.lifecycleConfig.language;
-    SDLLanguage actualLanguage = self.registerResponse.language;
-    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerShouldUpdateLifecycleToLanguage:)];
-    
+    // the language represents hmi language
+    SDLLanguage desiredHmiLanguage = self.configuration.lifecycleConfig.language;
+    SDLLanguage desiredVrLanguage = self.vrLanguage;;
+
+    SDLLanguage actualHmiLanguage = self.registerResponse.hmiDisplayLanguage;
+    SDLLanguage actualVrLanguage = self.registerResponse.language;
+
+    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerShouldUpdateLifecycleToLanguage:hmiLanguage:)];
+    SDLLogD(@"SDLLifecycleManager sdl_checkLanguageUpdating delegateCanUpdateLifecycle: %d", delegateCanUpdateLifecycle);
     // language mismatch? but actual language is a supported language? and delegate has implemented method?
-    if (![actualLanguage isEqualToEnum:desiredLanguage] && [supportedLanguages containsObject:actualLanguage] && delegateCanUpdateLifecycle) {
-        [self sdl_transitionToState:SDLLifecycleStateUpdatingConfiguration];
-    } else {
+    if (delegateCanUpdateLifecycle && [supportedLanguages containsObject:actualHmiLanguage] && [supportedLanguages containsObject:actualVrLanguage]) {
```

Comment:
This fires only if _both_ the HMI and VR language are supported. I'm thinking of a case where the app defaults to english, and the app supports english and Spanish. Then your head unit says that the HMI language is Spanish and the VR language is Japanese. This won't give the app an opportunity to update their language to Spanish. Is that intended behavior?